US09/warehouse

aaabramov opened the pull request:
- warehouse component
- warehouse user story (read only)
- warehouse MVC
- font awesome & gulp configs -> nav icons
- app name

ITsvetkoFF commented on src/client/app/warehouses/warehouses.controller.js:
```diff
@@ -0,0 +1,32 @@
+    activate();
+
+    function activate() {
+      var promises = [getWarehouses()];
```
so you're ready to add one more promise? Will you add it some day?

aaabramov replied:
yes,
promises for PATCH (when we want to put supplies)

ITsvetkoFF commented on src/client/app/warehouses/warehouses.route.js:
```diff
@@ -0,0 +1,32 @@
+          url: '/warehouses',
+          templateUrl: 'app/warehouses/warehouses.html',
+          controller: 'WarehousesController',
+          controllerAs: 'vm',
```
let's use controllerAs  more wise. each controller = unique namespace like `vmWarehouse`
can you please refactor all vm's in the project to meet this pattern?

aaabramov replied:
of course, it's my fault
I'm gonna refactor this now

ITsvetkoFF commented on src/client/app/warehouses/warehouses.route.js:
```diff
@@ -0,0 +1,32 @@
+          title: 'warehouses',
+          settings: {
+            nav: 3,
+            content: '<i class="fa fa-dashboard"></i>Warehouses'
```
Do you know what is `<i class="fa fa-dashboard"></i>` and why fontawesome does not work here? :) can you please fix?
(just use this solution https://github.com/ITsvetkoFF/Kv-013/pull/148/files)

ITsvetkoFF commented on src/client/app/warehouses/warehousesService.js:
```diff
@@ -0,0 +1,36 @@
+       • Move to constants
+       • route?
+       */
+      return $http.get('http://localhost:8080/api/v1/warehouses')
```
you CAN and SHOULD move because @onetwo1two  already pushed to develop constants (фуфуфу Вадим :) )

ITsvetkoFF commented on src/server/data.js:
```diff
@@ -4,13 +4,13 @@ module.exports = {
 
 function getPeople() {
   return [
-    { id: 1, firstName: 'John', lastName: 'Papa', age: 25, location: 'Florida' },
```
don't commit changes on files that we do not need :) only proper change is deletion :)

aaabramov replied:
Should I remove unnecessary components ?

aaabramov committed:
+ warehouse component
+ warehouse user story (read only)
+ warehouse MVC
+ font awesome & gulp configs -> nav icons
+ app name

ITsvetkoFF commented on bower.json:
```diff
@@ -27,7 +27,8 @@
     "angular-animate": "^1.5.3",
-    "angular-bootstrap": "^1.2.5"
+    "angular-bootstrap": "^1.2.5",
+    "components-font-awesome": "^4.6.1"
```
del font-awesome

ITsvetkoFF commented on gulp.config.js:
```diff
@@ -1,4 +1,4 @@
-module.exports = function() {
+module.exports = function () {
```
##

ITsvetkoFF commented on src/client/app/warehouses/warehouses.controller.js:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    function getWarehouses() {
+      return warehousesService.getWarehouses().then(function (data) {
```
move `warehousesService.getWarehouses().` to activate to get rid of this function

ITsvetkoFF commented on src/client/app/warehouses/warehouses.factory.js:
```diff
@@ -0,0 +1,32 @@
+
+  angular
+    .module('app.core')
+    .factory('warehousesService', warehousesService);
```
let is be warehouses
